### DISCUSSION TITLE
How do I enable translation in the android version?

### DISCUSSION THREAD

**Discussion by ivan89el** [diasurgical/DevilutionX#2976] [Q&A] [answered]
I did everything according to the instructions: https://github.com/diasurgical/devilutionX/wiki/Translation-Guide#step-3-testing
Placed ru.mo in /sdcard/android/data/org.diasurgical.devilutionx/files and Changed diablo.ini:
[Language]
Code=ru
But the language remained English.

**Comment by AJenbo** [comment, marked as answer]
Are you running a recent test build?

**Reply by ivan89el** [reply]
Yes I'm running the latest build

**Reply by AJenbo** [reply]
What do you belive to be the latest build?

**Reply by ivan89el** [reply]
4f43c39

**Reply by AJenbo** [reply]
Did you build it your self or download it?

**Reply by tsunamistate** [reply]
Did you copy devilutionx.mpq into /sdcard/android/data/org.diasurgical.devilutionx/files?
That contains the Cyrillic font

**Reply by ivan89el** [reply]
Downloaded. Yes I copied everything. The PC x86 build works with the same manipulations, and the language is enabled.

**Reply by StephenCWills** [reply]
I think Android doesn't load `.mo` files from external storage, and the test build doesn't have the `.gmo` files in the APK.

**Reply by AJenbo** [reply]
> Downloaded. Yes I copied everything. The PC x86 build works with the same manipulations, and the language is enabled.

Sorry but like @StephenCWills explained, the automatic Android test builds do not currently contain the translations, and for Android you cannot add them after the app is installed. Instead you would need to build the app your self using Android Studio, and make sure that you have gettext installed globally on your system.

The other option is to wait a few weeks for the 1.3.0 release that is coming in October, or potentially when ever we update the automatic Android test builds to include the translations.

Edit:
On a side note you can follow the Russian translation effort here: https://github.com/diasurgical/devilutionX/pull/2970

**Reply by ivan89el** [reply]
Thank you!
